sonyapilib/device.py

```diff
@@ -30,6 +30,42 @@
 WEBAPI_SERVICETYPE = "av:X_ScalarWebAPI_ServiceType"
 
 
+def xml_search_helper(data, param):
+    '''Performs find or findall on given xml with string from param.'''
+    if isinstance(param, (tuple, list)) and param[1] == "all":
```

Comment:
I think it would cleaner if the second parameter would a Boolean value instead of this string

Reply:
I think testing just for a truthy value makes sense, but in code it is easier to read if called with "all".

Reply:
Not quite sure about that because reading it for the time the association I've had was that the xml is queried for a string called "all" 
Maybe adding a third parameter called `read_all` and calling the method with this named parameter would be the cleanest way.

Reply:
with an additional parameter you are limited to one method (find OR findall).
i'm also okay with using True instead of "all". Just came in my mind while reading the code.

Reply:
I thought it wouldn't change too much always calling findall if it is needed in a call, but indeed if doing that it will fail pretty fast ...
That said I still prefer the boolean to prevent confusion with searching for an xml string.